Cython/Compiler/ExprNodes.py

```diff
@@ -2919,6 +2922,20 @@ def analyse_types(self, env):
             PyrexTypes.CFuncTypeArg("it", PyrexTypes.py_object_type, None),
             ]))
 
+    def has_reversed_cpp_iteration(self):
+        """
+        Returns True if the 'reversed' function is applied to a C++ iterable.
+
+        This supports C++ classes with reverse_iterator implemented.
+        """
+        reversed_cpp_iterable = False
+        if (isinstance(self.sequence, SimpleCallNode) and self.sequence.function.is_name and self.sequence.function.name == "reversed"):
```

Comment:
I see. I'll add something like
```python
if not self.sequence.function.entry.is_builtin:
  return False
```
I'm not pretty sure about the test. Is it going to be like this?
```python
def test_non_builtin_reversed_function(py_v):
    """
    >>> test_non_builtin_reversed_function([1, 3, 5])
    Non-buitin reversed called.
    5
    3
    1
    """
    def reversed(arg):
        print("Non-buitin reversed called.")
        return arg[::-1]

    cdef vector[int] vint
    for e in py_v:
        vint.push_back(e)
    for e in reversed(vint):
        print(e)
```

Reply:
You have almost exactly the right idea. I'd make the test so that your new `reversed` does something different and obvious (so that it's clear that it's being called instead of the built-in)